pipelines/tekton/resources.go

```diff
@@ -78,27 +78,33 @@ func generatePipeline(f fn.Function, labels map[string]string) *pplnv1beta1.Pipe
 	// Deploy step that uses an image produced by S2I builds needs explicit reference to the image
 	referenceImageFromPreviousTaskResults := false
 
+	var tasks []pplnv1beta1.PipelineTask
+	var buildPreReq []string
+
+	if f.Build.Git.URL != "" {
```

Comment:
This section would also benefit from an explicit directive (push vs. pull)

Reply:
wdym?

Reply:
Just that if the pipeline provider could be configured explicitly to deal with pulling from github OR pushing from the local host, this section would also benefit from that configuration.

Reply:
yeah, basically the stuff we discussed here: https://github.com/knative/func/pull/1298#discussion_r993530300